### ISSUE TITLE
Spinbox step and display less than 1

### ISSUE THREAD

**Issue by SGXander** [Nerwyn/service-call-tile-feature#25]
**Is your feature request related to a problem? Please describe.**
Hi, I'm using this to create a more succinct climate card and want to replicate the temperature spinbox but if I set the step to anything less than 1 (i.e. 0.5) the increment or decrement buttons do not work. similarly it does not display a decimal value e.g. 22.5

**Describe the solution you'd like**
I'd love to see support for decimal numbers to be displayed and stepped

**Describe alternatives you've considered**
The original tile feature can be used but I'd like to shorten my tile card to a couple of rows.

**Additional context**
N/A

**Comment by Nerwyn** [comment]
Definitely a bug, looking into it now.

**Comment by Nerwyn** [comment]
Turns out it was two bugs, one of which was in my ha-nunjucks project.
- Precision calculation was missing from spinboxes, causing their values to be treated as integers.
- `ha-nunjucks` was casting numerical strings to number, which was removing leading and trailing zeroes.
  - Fixed in [`ha-nunjucks` 1.2.3](https://github.com/Nerwyn/ha-nunjucks/releases/tag/1.2.3).

Can you try [3.3.2-beta.001](https://github.com/Nerwyn/service-call-tile-feature/releases/tag/3.3.2-beta.001)? It should fix your issue.

**Comment by SGXander** [comment]
wow thanks for the fast response! 
Confirmed 3.3.2-beta001 is working for me:
![image](https://github.com/Nerwyn/service-call-tile-feature/assets/8619536/eb0d2764-e086-4138-8bd3-8ef6131dad97)
with config:
```yml
type: tile
entity: climate.r07_c01_ac
name: Office AC
state_content: current_temperature
features:
  - type: climate-hvac-modes
  - type: custom:service-call
    haptics: true
    entries:
      - type: selector
        entity_id: climate.r07_c01_ac
        value_attribute: fan_mode
        options:
          - option: auto
            icon: mdi:fan-auto
            color: darkgreen
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: auto
          - option: diffuse
            icon: mdi:numeric-1
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: diffuse
          - option: low
            icon: mdi:numeric-2
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: low
          - option: medium
            icon: mdi:numeric-3
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: medium
          - option: middle
            icon: mdi:numeric-4
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: middle
          - option: high
            icon: mdi:numeric-5
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: high
      - type: spinbox
        value_attribute: temperature
        unit_of_measurement: ' °C'
        label: '{{ VALUE }}{{ UNIT }}'
        icon: mdi:thermometer
        step: '{{ state_attr("climate.r07_c01_ac", "target_temp_step") }}'
        debounceTime: 1000
        range:
          - '{{ state_attr("climate.r07_c01_ac", "min_temp") }}'
          - '{{ state_attr("climate.r07_c01_ac", "max_temp") }}'
        tap_action:
          action: call-service
          service: climate.set_temperature
          data:
            temperature: '{{ VALUE }}'
        style:
          flex-flow: row
          '--background': var(--disabled-color)
          '--icon-color': var(--tile-color)
  - type: custom:service-call
    haptics: true
    entries:
      - type: selector
        entity_id: select.r07_c01_vertical_vane
        options:
          - option: auto
            icon: mdi:alpha-a
            color: darkgreen
          - option: up
            icon: mdi:chevron-double-up
          - option: up_center
            icon: mdi:chevron-up
          - option: center
            icon: mdi:format-vertical-align-center
          - option: down_center
            icon: mdi:chevron-down
          - option: down
            icon: mdi:chevron-double-down
      - type: selector
        entity_id: select.r07_c01_horizontal_vane
        options:
          - option: left
            icon: mdi:chevron-double-left
          - option: left_center
            icon: mdi:chevron-left
          - option: center
            icon: mdi:format-horizontal-align-center
          - option: right_center
            icon: mdi:chevron-right
          - option: right
            icon: mdi:chevron-double-right
```

Small question: can config.entity be referenced as in mushroom? would save repeating on the entity_id

**Comment by Nerwyn** [comment]
> Small question: can config.entity be referenced as in mushroom? would save repeating on the entity_id

No, but I should look into adding something like that. The individual tile features don't have access to the entire tile config object, just their relative `entry` field. Still that includes it's `entity_id` and I can add the entire entry object (renamed to config) to the template context and remap `entity_id` to `entity`.

**Comment by SGXander** [comment]
As long as the entry is referenceable it would do great. I have a yaml based dashboard set for my room NSPanels and use yaml anchors to copy config where it appears frequently so this would be great for that alone.

I'll put it in as a separate issue for a feature request

**Comment by Nerwyn** [comment]
Keeping this open until the fix is officially released. It may be a few days as I'm busy with work/health/life for the next week.

**Comment by Nerwyn** [comment]
Then `entry` object is now available in templates as `config`, and the rendered template of `entity_id` is now available as `config.entity` in the latest alpha. There may be issues when using other entry/config fields that are templates as they aren't rendered yet, and nunjucks doesn't render recursively.

**Comment by Nerwyn** [comment]
@SGXander the latest 3.3.2 beta should be the release candidate. Let me know if it covers all of your issues!

**Comment by SGXander** [comment]
@Nerwyn Awesome thank you. can confirm it works beautifully for spinbox. Selector doesn't like '{{config.entity}}' though but that may be as you say above

**Comment by SGXander** [comment]
> @SGXander can you share your selector config? If the option value isn't a template it should work.

Of course:
```yml
type: tile
entity: climate.r07_c01_ac
name: Office Air Conditioner
state_content: current_temperature
features:
  - style: icons
    type: climate-hvac-modes
    hvac_modes:
      - 'off'
      - heat_cool
      - fan_only
      - cool
      - dry
      - heat
  - type: custom:service-call
    haptics: true
    entries:
      - type: selector
        entity_id: climate.r07_c01_ac     <<<<<<    '{{config.entity}}'  - produces null
        value_attribute: fan_mode
        options:
          - option: auto
            icon: mdi:fan-auto
            color: darkgreen
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: auto
          - option: diffuse
            icon: mdi:numeric-1
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: diffuse
          - option: low
            icon: mdi:numeric-2
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: low
          - option: medium
            icon: mdi:numeric-3
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: medium
          - option: middle
            icon: mdi:numeric-4
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: middle
          - option: high
            icon: mdi:numeric-5
            tap_action:
              action: call-service
              service: climate.set_fan_mode
              data:
                fan_mode: high
      - type: spinbox
        value_attribute: temperature
        unit_of_measurement: °C
        label: '{{ VALUE }} {{ UNIT }}'
        icon: mdi:thermometer
        icon_style:
          color: >-
            {% set ctemp = state_attr(config.entity, "current_temperature") %}
            {% set ttemp = state_attr(config.entity, "temperature") %} {% if
            ctemp < (ttemp - 2) %}
              blue
            {% elif ctemp < (ttemp - 1) and ctemp >= (ttemp - 2) %}
              cyan
            {% elif ctemp >= (ttemp - 1) and ctemp <= (ttemp + 1) %}
              green
            {% elif ctemp > (ttemp + 1) and ctemp <= (ttemp + 2) %}
              orange
            {% elif ctemp > (ttemp + 2) %}
              red
            {% endif %}
        step: '{{ state_attr(config.entity, "target_temp_step") }}'
        debounceTime: 1000
        range:
          - '{{ state_attr(config.entity, "min_temp") }}'
          - '{{ state_attr(config.entity, "max_temp") }}'
        tap_action:
          action: call-service
          service: climate.set_temperature
          data:
            temperature: '{{ VALUE }}'
        style:
          flex-flow: row
          '--background': var(--disabled-color)
  - type: custom:service-call
    haptics: true
    entries:
      - type: selector
        entity_id: select.r07_c01_vertical_vane
        options:
          - option: auto
            icon: mdi:alpha-a
            color: darkgreen
          - option: up
            icon: mdi:chevron-double-up
          - option: up_center
            icon: mdi:chevron-up
          - option: center
            icon: mdi:format-vertical-align-center
          - option: down_center
            icon: mdi:chevron-down
          - option: down
            icon: mdi:chevron-double-down
      - type: selector
        entity_id: select.r07_c01_horizontal_vane
        options:
          - option: left
            icon: mdi:chevron-double-left
          - option: left_center
            icon: mdi:chevron-left
          - option: center
            icon: mdi:format-horizontal-align-center
          - option: right_center
            icon: mdi:chevron-right
          - option: right
            icon: mdi:chevron-double-right
```
possibly to do with the value attribute but when doing the above it populates the selector but has no selected item and clicking one produces the below error:
![image](https://github.com/Nerwyn/service-call-tile-feature/assets/8619536/482ef202-340f-462f-9f04-53a6ca0eb87f)

If I skip the '' or {{}} it borks and doesnt show the selector at all

**Comment by Nerwyn** [comment]
Wait you're trying to populate the selector `entity_id` field using `{{ config.entity }}`. That's a circular argument! `config.entity` is the rendered template of `config.entity_id`. The feature's entity ID should be auto-populated using either the data/target entity ID or the tile card entity ID, unless `autofill_entity_id` is set to false. Try removing `entity_id` altogether.

**Comment by SGXander** [comment]
Ahhhh. Major derp moment then. removing the entity_id: def solves the problem. Sorry!!
Just need a way to derive the bottom two selectors from the parent now

**Comment by Nerwyn** [comment]
Those you'll have to explicitly set the feature entity ID for, as you have above. But the selector options should all take the selector feature entity ID unless you define it differently in the option or it's actions.

**Comment by SGXander** [comment]
yeah anything inside that field will reference itself now I follow you. my panel yaml will have to just be the top section for now then but that's fine as it's the biggest part anyway. thanks for being patient with me being daft!

**Comment by Nerwyn** [comment]
Fixed in [3.3.2](https://github.com/Nerwyn/service-call-tile-feature/releases/tag/3.3.2)
